[2024-06-01T16:07:13.484+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-06-01T16:07:13.647+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: etl_pipeline.extract_mssql_data scheduled__2023-01-13T00:00:00+00:00 [queued]>
[2024-06-01T16:07:13.695+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: etl_pipeline.extract_mssql_data scheduled__2023-01-13T00:00:00+00:00 [queued]>
[2024-06-01T16:07:13.698+0000] {taskinstance.py:2303} INFO - Starting attempt 1 of 2
[2024-06-01T16:07:13.765+0000] {taskinstance.py:2327} INFO - Executing <Task(PythonOperator): extract_mssql_data> on 2023-01-13 00:00:00+00:00
[2024-06-01T16:07:13.864+0000] {warnings.py:110} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61: DeprecationWarning: This process (pid=153) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-06-01T16:07:13.883+0000] {standard_task_runner.py:63} INFO - Started process 176 to run task
[2024-06-01T16:07:13.890+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'etl_pipeline', 'extract_mssql_data', 'scheduled__2023-01-13T00:00:00+00:00', '--job-id', '18', '--raw', '--subdir', 'DAGS_FOLDER/etl_dag.py', '--cfg-path', '/tmp/tmpwurwgzkk']
[2024-06-01T16:07:13.927+0000] {standard_task_runner.py:91} INFO - Job 18: Subtask extract_mssql_data
[2024-06-01T16:07:14.280+0000] {task_command.py:426} INFO - Running <TaskInstance: etl_pipeline.extract_mssql_data scheduled__2023-01-13T00:00:00+00:00 [running]> on host 773dc1cab949
[2024-06-01T16:07:14.782+0000] {taskinstance.py:2644} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='etl_pipeline' AIRFLOW_CTX_TASK_ID='extract_mssql_data' AIRFLOW_CTX_EXECUTION_DATE='2023-01-13T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2023-01-13T00:00:00+00:00'
[2024-06-01T16:07:14.793+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-06-01T16:07:14.916+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-06-01T16:07:14.928+0000] {taskinstance.py:2890} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 400, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/etl_dag.py", line 31, in extract_mssql_data
    mssql_extractor = MSSQLExtractor(config_file['mssql_connection_string'])
                                     ~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: '_io.TextIOWrapper' object is not subscriptable
[2024-06-01T16:07:15.024+0000] {taskinstance.py:1205} INFO - Marking task as UP_FOR_RETRY. dag_id=etl_pipeline, task_id=extract_mssql_data, execution_date=20230113T000000, start_date=20240601T160713, end_date=20240601T160715
[2024-06-01T16:07:15.138+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 18 for task extract_mssql_data ('_io.TextIOWrapper' object is not subscriptable; 176)
[2024-06-01T16:07:15.234+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-06-01T16:07:15.470+0000] {taskinstance.py:3482} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-06-01T16:07:15.478+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
